src/librustdoc/html/static/js/search.js

```diff
@@ -133,34 +124,359 @@ window.initSearch = function(rawSearchIndex) {
+                arr.splice(i, 1);
+                i -= 1;
+            }
+        }
+    }
+    function itemTypeFromName(typename) {
+        for (var i = 0, len = itemTypes.length; i < len; ++i) {
+            if (itemTypes[i] === typename) {
+                return i;
+            }
+        }
+        return NO_TYPE_FILTER;
+    }
+
+    function getStringElem(query, parserState, isInGenerics) {
+        if (isInGenerics) {
+            throw new Error("`\"` cannot be used in generics");
+        } else if (query.literalSearch) {
+            throw new Error("Cannot have more than one literal search element");
+        } else if (parserState.totalElems !== 0) {
+            throw new Error("Cannot use literal search when there is more than one element");
+        }
+        parserState.pos += 1;
+        while (parserState.pos < parserState.length && query.userQuery[parserState.pos] !== "\"") {
+            if (query.userQuery[parserState.pos] === "\\") {
+                // We ignore the next coming character.
+                parserState.pos += 1;
+            }
+            parserState.pos += 1;
+        }
+        if (parserState.pos >= parserState.length) {
+            throw new Error("Unclosed `\"`");
+        }
+        // To skip the quote at the end.
+        parserState.pos += 1;
+        query.literalSearch = true;
+    }
+    function skipWhitespaces(query, parserState) {
+        while (parserState.pos < parserState.length) {
+            var c = query.userQuery[parserState.pos];
+            if (!isWhitespace(c)) {
+                break;
+            }
+            parserState.pos += 1;
+        }
+    }
+    function skipStopCharacters(query, parserState) {
+        while (parserState.pos < parserState.length) {
+            var c = query.userQuery[parserState.pos];
+            if (!isStopCharacter(c)) {
+                break;
+            }
+            parserState.pos += 1;
+        }
+    }
+    function isPathStart(query, parserState) {
+        return parserState.pos + 1 < parserState.length &&
+            query.userQuery[parserState.pos] === ':' &&
+            query.userQuery[parserState.pos + 1] === ':';
+    }
+    function isReturnArrow(query, parserState) {
+        return parserState.pos + 1 < parserState.length &&
+            query.userQuery[parserState.pos] === '-' &&
+            query.userQuery[parserState.pos + 1] === '>';
+    }
+    function createQueryElement(query, parserState, elems, name, generics) {
+        removeEmptyStringsFromArray(generics);
+        if (name === '*' || (name.length === 0 && generics.length === 0)) {
+            return;
+        }
+        if (query.literalSearch && parserState.totalElems > 0) {
+            throw new Error("You cannot have more than one element if you use quotes");
+        }
+        var paths = name.split("::");
+        removeEmptyStringsFromArray(paths);
+        // In case we only have something like `<p>`, there is no name but it remains valid.
```

Comment:
I don't think we should support type parameters without a type name. It should be an error to search for, e.g. `<u8>`.

Reply:
That could be a doc alias, so not possible to throw an error for it. :-/

Reply:
We'll need to handle this in a follow-up then because of doc aliases.

Reply:
After thinking about it for some time, I think it's perfectly fine to have generics without a type in case you want to look for something in generics specifically.

Reply:
Resolving this thread in favor of the conversation at https://github.com/rust-lang/rust/pull/90630/files#r830507278, where we decided it was invalid to specify type parameters without a parent type (e.g. `<Borrow>`)